### PR TITLE
doc: ext: Extend zephyr-app-commands with new args

### PR DESCRIPTION
**Pull request by carlescufi**
New arguments added:

- conf: -DCONF_FILE=<>
- gen-args: Additional arguments to pass to CMake
- build-args: Additional arguments to pass to Make or Ninja

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

### CODE REVIEW

**Review by SebastianBoe — APPROVED** [review]

**Comment by carlescufi** [comment]
@mbolivar @jukkar I added an additional :compact: option that I personally plan to use in some parts of the doc

**Comment by carlescufi** [comment]
@mbolivar @jukkar Added a "run" goal for those that want a 2-step make and make run

File: doc/extensions/zephyr/application.py [hunk 1]
```diff
@@ -115,22 +120,25 @@ def run(self):
...
         elif app:
             content.append('$ cd {}'.format(app))
-            content.append('')
+            if not compact:
+                content.append('')
 
         if skip_config:
             content.append("# If you already made a build directory ({}) and ran cmake, just 'cd {}' instead.".format(build_dir, build_dir))  # noqa: E501
-        else:
+        elif not compact:
```
Comment (mbolivar) [inline doc/extensions/zephyr/application.py]: This doesn't behave as documented: if `:maybe-skip-config:` and `:compact:` are both set, the comment generated by maybe-skip-config is still present, even though compact says no comments will be present in the output text.

It seems that an error should be raised if `skip_config and compact`, as  they can't both win, and the behavior should be adjusted so the one which is present (if either) takes effect.

**Review by mbolivar — CHANGES_REQUESTED** [review]
These changes make sense to me, but the interaction between compact and maybe-skip-config needs a small adjustment.

**Review by mbolivar — APPROVED** [review]
Thanks! LGTM

**Review by jukkar — APPROVED** [review]
Works fine for me.

**Review by nashif — APPROVED** [review]

**Pull request merged**